internal/events/kafka.go

```diff
@@ -0,0 +1,115 @@
+package events
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/segmentio/kafka-go"
+	"github.com/stellar/go/support/log"
+	"golang.org/x/exp/maps"
+)
+
+type KafkaEventManager struct {
+	handlers []EventHandler
+	writer   *kafka.Writer
+	reader   *kafka.Reader
+}
+
+func NewKafkaEventManager(brokers []string, consumerTopics []string, consumerGroupID string) (*KafkaEventManager, error) {
+	k := KafkaEventManager{}
+
+	writer := kafka.NewWriter(kafka.WriterConfig{
+		Brokers:      brokers,
+		Balancer:     &kafka.RoundRobin{},
+		RequiredAcks: -1,
```

Comment:
Good call. Here is the producer's config. It means it will ensure the message was written on all brokers/partitions before returning. If it wasn't written, it returns an error. For the consumer, we use the consume message and after processing it we ACK it.

The library we use ([kafka-go](https://github.com/segmentio/kafka-go)), we have two methods to consume messages:

- `ReadMessage` - Reads the message and commits the offset.
- `FetchMessage` - Reads the message and does not commit the offset. We must use the `CommitMessages` method to commit it explicitly.

The approach I've chosen was the `FetchMessage` one.